generator/internal/scaffold_generator.cc

```diff
@@ -481,7 +506,513 @@ install(
+  auto constexpr kText = R"""(# Copyright $copyright_year$ Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is a minimal Makefile to show how to use the $title$ C++ client
+# for developers that use make(1) as their build system.
+
+# The CXX, CXXFLAGS and CXXLD variables are hard-coded. These values work for
+# our tests, but applications would typically make them configurable parameters.
+CXX=g++
+CXXFLAGS=
+CXXLD=$$(CXX)
+BIN=.
+
+all: $$(BIN)/quickstart
+
+# Configuration variables to compile and link against the $title$ C++
+# client library.
+CLIENT_MODULE     := google_cloud_cpp_$library$
+CLIENT_CXXFLAGS   := $$(shell pkg-config $$(CLIENT_MODULE) --cflags)
+CLIENT_CXXLDFLAGS := $$(shell pkg-config $$(CLIENT_MODULE) --libs-only-L)
+CLIENT_LIBS       := $$(shell pkg-config $$(CLIENT_MODULE) --libs-only-l)
+
+$$(BIN)/quickstart: quickstart.cc
+	$$(CXXLD) $$(CXXFLAGS) $$(CLIENT_CXXFLAGS) $$(CLIENT_CXXLDFLAGS) -o $$@ $$^ $$(CLIENT_LIBS)
+)""";
+  google::protobuf::io::OstreamOutputStream output(&os);
+  google::protobuf::io::Printer printer(&output, '$');
+  printer.Print(variables, kText);
+}
+
+void GenerateQuickstartWorkspace(
+    std::ostream& os, std::map<std::string, std::string> const& variables) {
+  // The version and SHA are hardcoded in this template, but it does not matter
+  // too much, renovate bot would update it as soon as it is merged.
+  auto constexpr kText = R"""(# Copyright $copyright_year$ Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A minimal WORKSPACE file showing how to use the $title$ C++
+# client library in Bazel-based projects.
+workspace(name = "$library$_quickstart")
+
+# Add the necessary Starlark functions to fetch google-cloud-cpp.
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+# Fetch the Google Cloud C++ libraries.
+# NOTE: Update this version and SHA256 as needed.
```

Comment:
hm. this is a little weird. but I think renovate-bot would update it.

Reply:
Yeah, I figured that the benefit of trying to generate a perfect `WORKSPACE` file was not worth it since it will get updated anyway.